### PR TITLE
Feat/header component v2

### PR DESCRIPTION
**Pull request by ErikssonJoakim**
This PR implements #319 

Parameter:
- `firstElement: JSX.Element` takes any element and is restricted to conform to component layout

There is a known bug where the text "BotAnik Space" does not work with light theme, see #333

### CODE REVIEW

File: src/ui/molecules/header/header.scss [hunk 1]
```diff
@@ -17,14 +17,20 @@
...
 
-  .okp4-header-logo-container {
-    grid-area: logo;
+  .okp4-header-first-element-container {
+    grid-area: firstElement;
+
+    :first-child {
+      max-width: 270px;
+      max-height: 72px;
+      display: flex;
```
Comment (fredericvilcot) [inline src/ui/molecules/header/header.scss:32]: What's the purpose of a `flex` display here?
We cannot deduce from the positioning of the elements passed as component property, it must remain the responsibility of the parent.

File: stories/molecules/components/header/Header.stories.mdx [hunk 2]
```diff
@@ -27,27 +32,65 @@ import './header.scss'
...
+- A `firstElement` that positions itself first in the reading flow. Typically used for logos. Will resize if larger than max size.
 - A `ThemeSwitcher` which allows each application to automatically manage the themes offered by the ØKP4 design system (Blue Stone / Light Stone).
 
 ## Overview
 
-<Story name="Overview">
-  <div style={{ width: '100%' }}>
-    <Header />
-  </div>
+<Story
```
Comment (fredericvilcot) [inline stories/molecules/components/header/Header.stories.mdx:50]: Please, add a code snippet associated to the story to help the user

File: stories/molecules/components/header/Header.stories.mdx [hunk 3]
```diff
@@ -27,27 +32,65 @@ import './header.scss'
...
+        type: { summary: 'JSX.Element' }
+      }
+    }
+  }}
+>
+  <Header firstElement={renderLogo()} />
 </Story>
 
+export const renderLogo = () => {
+  const { isXSmall } = useBreakpoint()
```
Comment (fredericvilcot) [inline stories/molecules/components/header/Header.stories.mdx:65]: This must be handled by the component itself, it should not be the responsibility of the parent to add a media query regarding the size of the viewport.

File: stories/molecules/components/header/Header.stories.mdx [hunk 4]
```diff
@@ -27,27 +32,65 @@ import './header.scss'
...
 ---
 
+### Properties
+
+<ArgsTable story="Overview" />
+
+## First element
+
+**firstElement** allows for any element to be put in the `header`.  
+There is a `max-width:270px` and a `max-height:72px`
```
Comment (fredericvilcot) [inline stories/molecules/components/header/Header.stories.mdx:78]: Sizes must be responsive, according to my comment above.

**Review by fredericvilcot — CHANGES_REQUESTED** [review]
Great job @ErikssonJoakim!
I wrote down some remarks, we can see together or with @lolottetheclash if you need ;)

File: src/ui/molecules/header/Header.tsx [hunk 5]
```diff
@@ -1,18 +1,19 @@
...
 import './header.scss'
-import { Logo } from '../../atoms/logo/Logo'
 import { ThemeSwitcher } from 'ui/atoms/theme/ThemeSwitcher'
-import type { Breakpoints } from 'hook/useBreakpoint'
-import { useBreakpoint } from 'hook/useBreakpoint'
+import type { DeepReadonly } from 'superTypes'
 
-export const Header: React.FC = (): JSX.Element => {
-  const { isXSmall }: Breakpoints = useBreakpoint()
+export type HeaderProps = DeepReadonly<{
```
[the remark this answers is not in the document]
Comment (ErikssonJoakim) [reply src/ui/molecules/header/Header.tsx:6]: Oki I will correct this 👍

File: src/ui/molecules/header/header.scss (hunk 1 above)
Comment (ErikssonJoakim) [reply src/ui/molecules/header/header.scss:32]: The purpose is that the container is too small to realistically have another layout. But as you say we shouldn't deduce the layout of the property so I will remove this

**Review by fredericvilcot — CHANGES_REQUESTED** [review]
You're almost there ;)
Just a few tweaks, keep going!

File: src/ui/molecules/header/Header.tsx [hunk 6]
```diff
@@ -1,17 +1,25 @@
...
+export const Header: React.FC<HeaderProps> = ({
+  firstElement
+}: DeepReadonly<HeaderProps>): JSX.Element => {
+  const { isXSmall }: Breakpoints = useBreakpoint()
   return (
-    <div className={`okp4-header-main`}>
-      <div className="okp4-header-logo-container">
-        <Logo size={isXSmall ? 'small' : 'medium'} />
+    <div className="okp4-header-main">
+      <div className={`okp4-header-first-element-container ${isXSmall ? 'is-x-small' : ''}`}>
```
[the remark this answers is not in the document]
Comment (ErikssonJoakim) [reply src/ui/molecules/header/Header.tsx:21]: Alright I'll change it

File: stories/molecules/components/header/Header.stories.mdx [hunk 7]
```diff
@@ -27,27 +31,63 @@ import './header.scss'
...
 
 [![stability-unstable](https://img.shields.io/badge/stability-unstable-yellow.svg)](https://github.com/emersion/stability-badges#unstable)
 
 ## Description
 
-**Header component** is a responsive component which embeds two main elements:
+**Header** is a responsive component which embeds two main elements:
 
-- A `Logo` which must be part of the set of applications representing the ØKP4 brand.
+- A `firstElement` that positions itself first in the reading flow. Typically used for logos. Will resize if larger than max size.
```
[the remark this answers is not in the document]
Comment (ErikssonJoakim) [reply stories/molecules/components/header/Header.stories.mdx:44]: I would say the difference is subtle but I agree `comes` is more accurate

File: stories/molecules/components/header/Header.stories.mdx [hunk 8]
```diff
@@ -27,27 +31,63 @@ import './header.scss'
...
 
 ---
 
+### Properties
+
+<ArgsTable story="Overview" />
+
+## First element
+
+**firstElement** allows for any element to be put in the `header`.
```
[the remark this answers is not in the document]
Comment (ErikssonJoakim) [reply stories/molecules/components/header/Header.stories.mdx:71]: For the moment the property is not optional. I'll see what I can do write to clarify 😃

**Review by fredericvilcot — APPROVED** [review]
All good!

**Pull request merged**